components/TARGET_PSA/services/storage/ps/COMPONENT_NSPE/protected_storage.cpp

```diff
@@ -95,14 +95,20 @@ psa_status_t psa_ps_set(psa_storage_uid_t uid, uint32_t data_length, const void
     }
 
     uint32_t kv_create_flags = def_kvstore_flags;
+    if (create_flags & PSA_STORAGE_FLAG_NO_CONFIDENTIALITY) {
+        kv_create_flags &= ~KVStore::REQUIRE_CONFIDENTIALITY_FLAG;
+    }
+    if (create_flags & PSA_STORAGE_FLAG_NO_REPLAY_PROTECTION) {
+        kv_create_flags &= ~KVStore::REQUIRE_REPLAY_PROTECTION_FLAG;
+    }
```

Comment:
What if `def_kvstore_flags` is missing `REQUIRE_CONFIDENTIALITY_FLAG` and the PSA `create_flags` is missing `PSA_STORAGE_FLAG_NO_CONFIDENTIALITY`? Don't we want to enable additional bits in `def_kvstore_flags` to ensure that the PSA set that requires confidentiality would get it?

Since this is tricky logic, could you point me to the automated test that ensures this works correctly or create one if it doesn't exist yet?

Reply:
As explained above, the `PSA_STORAGE_FLAG_NO_CONFIDENTIALITY` was replaced to `PSA_STORAGE_FLAG_NO_REPLAY_PROTECTION`.

Reply:
This looks better now, thanks!